packages/ra-ui-materialui/src/input/DateInput.tsx

```diff
@@ -231,8 +229,10 @@ const getStringFromDate = (value: string | Date) => {
     }
 
     // valid dates should not be converted
-    if (dateRegex.test(value)) {
-        return value;
+    // This regex only returns the date part of the provided string
+    const matches = dateRegex.exec(value);
+    if (matches) {
+        return matches[1];
     }
 
     return convertDateToString(new Date(value));
```

Comment:
**question**: should we still attempt to call `convertDateToString` if the value does not match the regex? :thinking:

Reply:
Yes, because `Date` also handles exotic formats that do not match ISO 8601